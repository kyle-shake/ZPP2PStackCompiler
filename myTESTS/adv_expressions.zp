//simple-exp.zp
//input: none
//output: 5, 5, "15 mod 10 equals 5!",
//	  "15 divided by 3 also equals 5!"
//	  "15 is greater than 15 divided by 3"
//	  "15 is greater than or equal to 15"
//        1 \n "is less than or equal to 15"
//        3 \n "is less than or equal to 15"
//        7 \n "is less than or equal to 15"
//        15 \n "is less than or equal to 15"

int main()
{

   int num;
   num = 15;
   
   int mod;
   mod = num % 10;
   putn(mod);
   
   int div;
   div = num / 3;
   putn(div);

   if(mod == div){
       puts("Test Equivalency expression passed!\n");
       putn(mod);
       putn(div);
   }

   if(num > div){
       puts("Test Greater Than expression passed!\n");
       putn(num);
       putn(div);
   }

   if(num >= 15){
       puts("Test Greater Than or Equal To expression passed!\n");
       putn(num);
       putn(15);
   } 

   int i;
   i = 1;
   while(i <= num){
       puts("Test Less Than or Equal To expression passed!\n");
       putn(i);
       putn(num);
       i = 2 * i + 1;
   }

   if( i > div && i > mod){
       puts("Test And Expression passed!\n");
   }

   if( mod > num || i > num){
       puts("Test Or Expression passed!\n");
   }

   if( !(i == 0) ){
       puts("Test Not Expression with Equivalency passed!\n");
   }
 
   if(!(i < num)){
       puts("Test Not Expression with Greater than passed!\n");
   }

   return(0);
}
