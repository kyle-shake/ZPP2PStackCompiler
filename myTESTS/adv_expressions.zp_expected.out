     5 
     5 
Test Equivalency expression passed!
     5 
     5 
Test Greater Than expression passed!
    15 
     5 
Test Greater Than or Equal To expression passed!
    15 
    15 
Test Less Than or Equal To expression passed!
     1 
    15 
Test Less Than or Equal To expression passed!
     3 
    15 
Test Less Than or Equal To expression passed!
     7 
    15 
Test Less Than or Equal To expression passed!
    15 
    15 
Test And Expression passed!
Test Or Expression passed!
Test Not Expression with Equivalency passed!
Test Not Expression with Greater than passed!
